prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/azure/AzureCreateVirtualMachineTask.java

```diff
@@ -0,0 +1,147 @@
+			final String azureSubscriptionId = getRequiredParameterValue(context, "azure-subscription-id");
+			final String azureClientId = getRequiredParameterValue(context, "azure-client-id");
+			final String azureClientSecret = getRequiredParameterValue(context, "azure-client-secret");
+			final String resourcesPrefix = getRequiredParameterValue(context, "resources-prefix");
+
+			TokenCredential credential = new ClientSecretCredentialBuilder().tenantId(azureTenantId)
+					.clientId(azureClientId).clientSecret(azureClientSecret).build();
+
+			AzureProfile profile;
+			if (StringUtils.hasLength(azureTenantId) && StringUtils.hasLength(azureSubscriptionId)) {
+				profile = new AzureProfile(azureTenantId, azureSubscriptionId, AzureEnvironment.AZURE);
+			}
+			else {
+				profile = new AzureProfile(AzureEnvironment.AZURE);
+			}
+
+			AzureResourceManager azureResourceManager = AzureResourceManager.configure()
+					.withLogLevel(HttpLogDetailLevel.BASIC).authenticate(credential, profile).withDefaultSubscription();
+
+			log.info("Creating resource group...");
+			ResourceGroup resourceGroup = azureResourceManager.resourceGroups()
+					.define(resourcesPrefix + "ResourceGroup").withRegion(Region.US_EAST).create();
+
+			log.info("Creating availability set...");
+			AvailabilitySet availabilitySet = azureResourceManager.availabilitySets()
+					.define(resourcesPrefix + "AvailabilitySet").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").create();
+
+			log.info("Creating public IP address...");
+			PublicIpAddress publicIPAddress = azureResourceManager.publicIpAddresses()
+					.define(resourcesPrefix + "PublicIP").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").withDynamicIP().create();
+
+			log.info("Creating virtual network...");
+			Network network = azureResourceManager.networks().define(resourcesPrefix + "VN").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").withAddressSpace("10.0.0.0/16")
+					.withSubnet(resourcesPrefix + "Subnet", "10.0.0.0/24").create();
+
+			log.info("Creating network interface...");
+			NetworkInterface networkInterface = azureResourceManager.networkInterfaces().define(resourcesPrefix + "NIC")
+					.withRegion(Region.US_EAST).withExistingResourceGroup(resourcesPrefix + "ResourceGroup")
+					.withExistingPrimaryNetwork(network).withSubnet(resourcesPrefix + "Subnet")
+					.withPrimaryPrivateIPAddressDynamic().withExistingPrimaryPublicIPAddress(publicIPAddress).create();
+
+			log.info("Creating virtual machine...");
+			VirtualMachine virtualMachine = azureResourceManager.virtualMachines().define(resourcesPrefix + "VM")
+					.withRegion(Region.US_EAST).withExistingResourceGroup(resourcesPrefix + "ResourceGroup")
+					.withExistingPrimaryNetworkInterface(networkInterface)
+					.withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
```

Comment:
should we make this dynamic to support windows server or other systems?

Reply:
+1, I'd suggest to parameterized it (or a closed list of options) + also let the user specify VirtualMachineSizeTypes

Reply:
I suggest to have this change after the demo

Reply:
As from the first impression I got, its not just about changing the machine type, also the function (withPopularLinuxImage) and the size might be changed .
Need to investigate more about the types included.